packages/image/src/vite-plugin.js

```diff
@@ -0,0 +1,145 @@
+import { format, imagetools, resize } from 'vite-imagetools';
+
+/**
+ * @param {import('types/vite').PluginOptions} options
+ * @returns {import('vite').Plugin[]}
+ */
+export function vitePluginSvelteImage(options = {}) {
+	return [image_plugin(options), image_tools(options)];
+}
+
+/**
+ * Creates the Svelte image plugin which provides the `__svelte_image_options__` module used in `Image.svelte`.
+ * @param {import('types/vite').PluginOptions} options
+ * @returns {import('vite').Plugin}
+ */
+function image_plugin(options) {
+	if (!options.providers) {
+		console.warn(
+			'vite-plugin-svelte-image: No provider found for @sveltejs/image, images not optimized at build time will not be optimized'
+		);
+	}
+	const providers =
+		!options.providers || !Object.keys(options.providers).length
+			? { default: '@sveltejs/image/providers/none' }
+			: options.providers;
+
+	const providerImports = Object.keys(providers)
+		.map(
+			(provider) =>
+				`import * as ${provider === 'default' ? '_default' : provider} from '${
+					providers[provider]
+				}';`
+		)
+		.join('\n');
+	const providerObject =
+		'{\n\t' +
+		Object.keys(providers)
+			.map((provider) => `${provider}: ${provider === 'default' ? '_default' : provider}`)
+			.join(',\n\t') +
+		'\n}';
+	const file = `${providerImports}
+export const providers = ${providerObject};
+export const domains = ${options.domains ? JSON.stringify(options.domains) : '[]'};
+export const device_sizes = ${JSON.stringify(device_sizes(options))};
+export const image_sizes = ${JSON.stringify(image_sizes(options))};`;
+
+	return {
+		name: 'vite-plugin-svelte-image',
+		async resolveId(id) {
+			if (id === '__svelte-image-options__.js') {
+				return id;
+			}
+		},
+		async load(id) {
+			if (id === '__svelte-image-options__.js') {
+				return file;
+			}
+		}
+	};
+}
+
+/**
+ * Wraps `vite-imagetools` to provide a `generate` output format that returns an object with `src`, `width`, `height` and `srcset` properties.
+ * @param {import('types/vite').PluginOptions} options
+ */
+function image_tools(options) {
+	return imagetools({
+		include: /^[^?]+\.(heic|heif|avif|jpeg|jpg|png|tiff|webp|gif)\?(generate)$/,
+		extendOutputFormats: (builtins) => {
+			/** @type {import('vite-imagetools').OutputFormat} */
+			function generate() {
+				return (metadata) => {
+					return {
+						src: metadata[0].src,
+						width: metadata[0].width,
+						height: metadata[0].height,
```

Comment:
Let's use `w` and `h` here to align with `vite-imagetools`. It gets embedded in the JS and is smaller, so it's also a minor space savings
```suggestion
						w: metadata[0].width,
						h: metadata[0].height,
```